### ISSUE TITLE
Pandas transform output option is ignored or raises error

### ISSUE THREAD

**Issue by enricogandini** [EBjerrum/scikit-mol#38]
## Description

Starting from scikit-learn 1.2, it is possible to set the output of transformers to be Pandas objects (Series, DataFrame) instead of the default NumPy arrays: https://scikit-learn.org/stable/auto_examples/miscellaneous/plot_set_output.html#introducing-the-set-output-api
This can be done:
- globally by calling `sklearn.set_config(transform_output="pandas")`,
- or directly on a Transformer object (or Pipeline) with method `set_output(transform="pandas")`.

Scikit-mol ignores the globally-set option: the output of transformers is still NumPy arrays even after setting it to "pandas".

If `set_output` is called directly on a scikit-mol transformer object (like `SmilesToMolTransformer`), we get:
```
AttributeError: This 'SmilesToMolTransformer' has no attribute 'set_output'
```

If instead `set_output` is called on a Pipeline which includes scikit-mol transformer, we get:
```
ValueError: Unable to configure output for SmilesToMolTransformer() because `set_output` is not available.
```

## Solution

Make scikit-mol transformers compatible with the `set_output` API.
This can be achieved by defining `set_output`, `get_feature_names_out`, and `auto_wraps_output_keys` on the scikit-mol transformer classes (or their base class, like `FpsTransformer`). See [this](https://github.com/scikit-learn/scikit-learn/blob/4e8253703013b38da503e2354d82fb7fa43dd4ec/sklearn/utils/_set_output.py#L332) for reference.

For instance, the output of a `MorganFingerprintTransformer` with `nBits=1024` could be a DataFrame with column names `"fp_morgan_0001"`, `"fp_morgan_0002"`, ..., `"fp_morgan_1024"`.

**Comment by enricogandini** [comment]
I am trying to solve this issue, I am working on this fork: https://github.com/enricogandini/scikit-mol

**Comment by EBjerrum** [comment]
Thanks for bringing this to our attention. With big fingerprint size, it's going to be some massive dataframes, but with for instance the descriptor transformer it will make super good sense to have named columns. Thanks for looking into this ;-) We have a slack channel if you want to join. It's not super active, but if you want to discuss it.

**Comment by enricogandini** [comment]
Thanks @EBjerrum, I would like to join. I also have some other little features in mind, and I would like to discuss them.

**Comment by EBjerrum** [comment]
Sure, sounds good, what email should I send the invite to?

**Comment by enricogandini** [comment]
To ***@***.*** , thanks!

Il Mar 27 Feb 2024, 09:06 Esben Jannik Bjerrum ***@***.***>
ha scritto:

**Comment by EBjerrum** [comment]
This was solved with the v0.3
